### ISSUE TITLE
MULTIPLE_SUBSCRIPTIONS | repeat supscription

### ISSUE THREAD

**Issue by ronag** [deepstreamIO/deepstream.io-client-js#147]
I'm getting the following error on the server side:

```
S | for R:media/cioznw85i0001ouvic5vlqrzu by undefined
S | for R:file/ac30c9639a0c4d83f066d379c8cb87fb by undefined
US | for R:file/ac30c9639a0c4d83f066d379c8cb87fb by undefined
US | for R:media/cioznw85i0001ouvic5vlqrzu by undefined
S | for R:media/cioznw85i0001ouvic5vlqrzu by undefined
S | for R:file/ac30c9639a0c4d83f066d379c8cb87fb by undefined
US | for R:file/ac30c9639a0c4d83f066d379c8cb87fb by undefined
US | for R:media/cioznw85i0001ouvic5vlqrzu by undefined
S | for R:media/cioznw85i0001ouvic5vlqrzu by undefined
S | for R:file/ac30c9639a0c4d83f066d379c8cb87fb by undefined
US | for R:file/ac30c9639a0c4d83f066d379c8cb87fb by undefined
US | for R:media/cioznw85i0001ouvic5vlqrzu by undefined
S | for R:media/cioznw85i0001ouvic5vlqrzu by undefined
MULTIPLE_SUBSCRIPTIONS | repeat supscription to "media/cioznw85i0001ouvic5vlqrzu" by undefined
S | for R:file/ac30c9639a0c4d83f066d379c8cb87fb by undefined
US | for R:file/ac30c9639a0c4d83f066d379c8cb87fb by undefined
US | for R:media/cioznw85i0001ouvic5vlqrzu by undefined
S | for R:media/cioznw85i0001ouvic5vlqrzu by undefined
S | for R:file/ac30c9639a0c4d83f066d379c8cb87fb by undefined
US | for R:file/ac30c9639a0c4d83f066d379c8cb87fb by undefined
US | for R:media/cioznw85i0001ouvic5vlqrzu by undefined
S | for R:media/cioznw85i0001ouvic5vlqrzu by undefined
S | for R:file/ac30c9639a0c4d83f066d379c8cb87fb by undefined
US | for R:file/ac30c9639a0c4d83f066d379c8cb87fb by undefined
US | for R:media/cioznw85i0001ouvic5vlqrzu by undefined
S | for R:media/cioznw85i0001ouvic5vlqrzu by undefined
MULTIPLE_SUBSCRIPTIONS | repeat supscription to "media/cioznw85i0001ouvic5vlqrzu" by undefined
S | for R:file/ac30c9639a0c4d83f066d379c8cb87fb by undefined
US | for R:file/ac30c9639a0c4d83f066d379c8cb87fb by undefined
US | for R:media/cioznw85i0001ouvic5vlqrzu by undefined
S | for R:media/cioznw85i0001ouvic5vlqrzu by undefined
```

and on the client side:

```
12:02:55.623Z  INFO : Deepstream Error. (error=media/cioznw85i0001ouvic5vlqrzu, event=UNSOLICITED_MESSAGE, topic=R)
12:02:55.623Z  INFO : Deepstream Error. (error=media/cioznw85i0001ouvic5vlqrzu, event=UNSOLICITED_MESSAGE, topic=R)
12:02:56.622Z  INFO : Deepstream Error. (error=media/cioznw85i0001ouvic5vlqrzu, event=ACK_TIMEOUT, topic=R)
12:02:56.622Z  WARN : ACK_TIMEOUT
12:02:58.626Z  INFO : Deepstream Error. (error=media/cioznw85i0001ouvic5vlqrzu, event=RESPONSE_TIMEOUT, topic=R)
12:02:58.626Z  WARN : RESPONSE_TIMEOUT
```

**Comment by ronag** [comment]
One possible reason might be that the client forcefully quits sometimes.

**Comment by ronag** [comment]
It seems that rapidly subscribing/unsubscribing/killing can cause some funkiness. I'll try to get make a smaller reproducible sample.

**Comment by yasserf** [comment]
Be great if you can give us a repo to reproduce this issue!

**Comment by yasserf** [comment]
any chance u were able to reproduce this issue on demand?

**Comment by ronag** [comment]
@yasserf: I haven't had time to do that yet. We've worked around the issue by keeping a local cache of record to avoid excessive sub/unsub.

**Comment by yasserf** [comment]
1.0.2 fixed an issue with repeat subscriptions that was due to deleted. Can I confirm you didn't do anything with deletes in this example?

**Comment by ronag** [comment]
@yasserf: No deletes.

**Comment by yasserf** [comment]
This should be fixed in 1.0.2
